File: SPCServer/springboot/src/main/java/com/lcaohoanq/shoppe/domain/category/Subcategory.java
package com.lcaohoanq.shoppe.domain.category;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.lcaohoanq.shoppe.base.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "subcategories")
@JsonPropertyOrder({
    "id",
    "name",
    "category"
})
public class Subcategory extends BaseEntity implements Comparable<Subcategory> {

    @Id
    @SequenceGenerator(name = "subcategories_seq", sequenceName = "subcategories_id_seq", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "subcategories_seq")
    @Column(name="id", unique=true, nullable=false)
    @JsonProperty("id")
    private Long id;
    private String name;
    
    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "category_id")
    private Category category;

    @Override
    public int compareTo(Subcategory that) {
        return this.id.compareTo(that.id);
    }
}

File: product-service/src/main/kotlin/com/lcaohoanq/productservice/domain/category/SubCategory.kt
package com.lcaohoanq.productservice.domain.category

import BaseEntity
import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*

@Entity
@Table(name = "subcategories")
class SubCategory: BaseEntity(), Comparable<SubCategory> {

    @Id
    val id: Long? = null
    val name: String? = null

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    @JsonIgnore
    val category: Category = Category()

    override fun compareTo(other: SubCategory): Int {
        return compareValues(this.id, other.id)
    }

}